editor/plugins/tiles/tile_set_atlas_source_editor.cpp

```diff
@@ -2242,54 +2246,61 @@ void TileSetAtlasSourceEditor::edit(Ref<TileSet> p_tile_set, TileSetAtlasSource
 	_update_current_tile_data_editor();
 }
 
-void TileSetAtlasSourceEditor::init_source() {
+void TileSetAtlasSourceEditor::init_source(Vector<Ref<TileSetAtlasSource>> atlases) {
 	tool_setup_atlas_source_button->set_pressed(true);
+	atlases_to_auto_create_tiles = atlases;
 	confirm_auto_create_tiles->popup_centered();
 }
 
 void TileSetAtlasSourceEditor::_auto_create_tiles() {
-	if (!tile_set_atlas_source) {
-		return;
-	}
+	for (Ref<TileSetAtlasSource> atlas_source : atlases_to_auto_create_tiles) {
```

Comment:
```suggestion
	for (Ref<TileSetAtlasSource> &atlas_source : atlases_to_auto_create_tiles) {
```
Reference here, no need to refcount